wire-ios-data-model/Source/Model/Confirmation/ZMMessageConfirmation.swift

```diff
@@ -22,7 +22,7 @@ import Foundation
 @objc public enum MessageConfirmationType: Int16 {
     case delivered, read
 
-    static func convert(_ zmConfirmationType: Confirmation.TypeEnum) -> MessageConfirmationType {
+    public static func convert(_ zmConfirmationType: Confirmation.TypeEnum) -> MessageConfirmationType {
```

Comment:
Many legacy methods were made public to facilitate the work in this PR (not ideal but could be adjusted later on). Conversely, a lot of existing logic was factored out when we had no other option (e.g some legacy methods would take a `ZMUpdatedEvent` as parameter).